clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -3008,11 +3008,14 @@ bool CIRGenModule::supportsCOMDAT() const {
   return getTriple().supportsCOMDAT();
 }
 
-void CIRGenModule::maybeSetTrivialComdat(const Decl &D, mlir::Operation *Op) {
-  if (!shouldBeInCOMDAT(*this, D))
+void CIRGenModule::maybeSetTrivialComdat(const Decl &d, mlir::Operation *op) {
+  if (!shouldBeInCOMDAT(*this, d))
     return;
-
-  // TODO: Op.setComdat
+  auto globalOp = dyn_cast_or_null<mlir::cir::GlobalOp>(op);
+  if (globalOp)
+    globalOp.setComdat(true);
+  // Keep it as missing feature as we need to implement comdat for FuncOp.
+  // in the future.
   assert(!MissingFeatures::setComdat() && "NYI");
```

Comment:
Should this assert be removed?

Reply:
Oh, nevermind!

Reply:
We haven't implemented comdat for FuncOp yet, also 
[vtable ](https://github.com/llvm/clangir/blob/8710834c4b7563e88521759eb98751fe3ca82d9b/clang/lib/CIR/CodeGen/CIRGenVTables.cpp#L521)